### DISCUSSION TITLE
import ”UNet2DConditionOutput“ error

### DISCUSSION THREAD

**Discussion by txy00001** [huggingface/diffusers#8866] [Q&A]
pip install --upgrade diffusers[torch]     The error message "No module named 'diffusers.models.unet_2d_condition'" is in the source code, it is imported in models/unet, but according to the installation method, the installed diffusers are imported by models.unet_2d_condition, but the latest source code does not match, as if there is no corresponding version update

**Comment by txy00001** [comment]
The version in the source code is: 0.30.0.dev0, but the version is 0.29.2 when installed with pip install --upgrade diffusers[torch], which causes the UNet2DConditionOutput to be imported incorrectly, and the version update does not match

**Reply by asomoza** [reply]
Hi, this is expected, installing from source is for people that wants to test the latest changes that haven't been released and can be unstable and also for contributors.

Installing from pip is for other libraries and people that just want to use the more stable version of diffusers, you can't upgrade from the released versions to the source one.

The unet refactor was made in this [PR](https://github.com/huggingface/diffusers/pull/6630) which was merged in january and was included in the [0.26 release](https://github.com/huggingface/diffusers/releases/tag/v0.26.0). 

Since you're having an error with this, means that you either somehow have a cached pre 0.26 version in your environment, or you're using external code that has the old imports.

A difference between 0.29.2 and 0.30.0.dev0 wouldn't bring that issue.

**Comment by LukeLIN-web** [comment]
Meet the same problem  In diffusers-0.31.0

**Comment by ashishsethi20** [comment]
facing the same issue, tried installing all versions from 0.28.2 to latest one
